OAuth Login Sample/swift/OAuthLogin/Controllers/ProfileViewController.swift

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2015 Esri
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import UIKit
+import ArcGIS
+
+class ProfileViewController: UIViewController, AGSPortalUserDelegate {
+
+    @IBOutlet weak var thumbnailView:UIImageView!
+    
+    @IBOutlet weak var fullNameTextField:JVFloatLabeledTextField!
+    @IBOutlet weak var usernameTextField:JVFloatLabeledTextField!
+    @IBOutlet weak var emailTextField:JVFloatLabeledTextField!
+    @IBOutlet weak var memberSinceTextField:JVFloatLabeledTextField!
+    @IBOutlet weak var roleTextField:JVFloatLabeledTextField!
+    @IBOutlet weak var bioTextField:JVFloatLabeledTextField!
+    
+    @IBOutlet weak var signOutButton:UIButton!
+    
+    var portal:AGSPortal!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        //disable back button
+        self.navigationItem.hidesBackButton = true
+        
+        //add corner radius and border for the thumbnail
+        self.thumbnailView.layer.cornerRadius = self.thumbnailView.bounds.size.width/2
+        self.thumbnailView.layer.borderColor = UIColor.whiteColor().CGColor
+        self.thumbnailView.layer.borderWidth = 2
+        
+        //add corner radius for the button
+        self.signOutButton.layer.cornerRadius = self.signOutButton.bounds.size.height/2
+        
+        //populate the data using the user object on portal
+        self.populateData()
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+    
+    func populateData() {
+        //if thumbnail already loaded then simply assign it
+        //else initiate the fetch
+        if self.portal.user.thumbnail != nil {
+            self.thumbnailView.image = self.portal.user.thumbnail
+        }
+        else {
+            self.portal.user.delegate = self
+            self.portal.user.fetchThumbnail()
```

Comment:
check if the thumbnail exists (`thumbnailFileName` is not nil) before fetching it, otherwise the delegate reports an unnecessary error while fetching a thumbnail that does not exist.